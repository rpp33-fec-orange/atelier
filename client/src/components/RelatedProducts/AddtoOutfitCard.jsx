import React from 'react';

const rowStyle = {
  // display: 'flex'
};

const columnStyle = {
  // flex: '33.33%',
  // padding: '5px',
  // border: '1px solid #555',
  // width: '150'
};

const addToOutfit = 'https://images.cdn1.stockunlimited.net/preview1300/plus-sign_1647891.jpg'

const AddtoOutfitCard = (props) => {

    return (
        <div className = "your-outfit-card">
          <img src = {addToOutfit} alt = "Sample Image" height = "150" width = '150' onClick = {(e) => {props.handleClick(e)}}/>
          <div className = "add-to-outfit-button" onClick = {(e) => {props.handleClick(e)}} height = "150" width = '150'></div>
        </div>
    )

}

export default AddtoOutfitCard;