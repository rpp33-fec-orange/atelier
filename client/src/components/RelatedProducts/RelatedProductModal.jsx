import React from 'react';

class RelatedProductModal extends React.Component {

  constructor(props) {
    super(props)
    this.state = {
      combinedFeatures: []
    }
    this.onClose = this.onClose.bind(this);
    this.combinedFeatures = this.combinedFeatures.bind(this);
  }

  onClose(e) {
    this.props.onClose() && this.props.onClose(e);
  }

  combinedFeatures() {

    const {parentProduct, relatedProduct} = this.props;
    var parentProductFeatures = parentProduct.features;
    var relatedProductFeatures = relatedProduct.features;
    var combinedFeatures = parentProductFeatures.concat(relatedProductFeatures);

    for (var i = 0; i < combinedFeatures.length; i++) {
      for (var j = 0; j < parentProductFeatures.length; j++) {
        if (combinedFeatures[i].feature === parentProductFeatures[j].feature) {
          if (combinedFeatures[i].value === null) {
            combinedFeatures[i].parentProductValue = '';
            combinedFeatures[i].relatedProductValue = '';
          } else if (combinedFeatures[i].value === true) {
            combinedFeatures[i].parentProductValue = '✔️';
            combinedFeatures[i].relatedProductValue = '';
          } else {
            combinedFeatures[i].parentProductValue = combinedFeatures[i].value;
            combinedFeatures[i].relatedProductValue = '';
          }
        }
      }
    }

    for (var i = 0; i < combinedFeatures.length; i++) {
      for (var j = 0; j < relatedProductFeatures.length; j++) {

        if (combinedFeatures[i].parentProductValue === undefined) {
          combinedFeatures[i].parentProductValue = '';
        }
        if (combinedFeatures[i].feature === relatedProductFeatures[j].feature) {
          if (combinedFeatures[i].value === null) {
            combinedFeatures[i].relatedProductValue = '';
          } else if (combinedFeatures[i].value === true) {
            combinedFeatures[i].relatedProductValue = '✔️';
          } else {
            combinedFeatures[i].relatedProductValue = combinedFeatures[i].value;
          }
        }
      }
    }
    return combinedFeatures;
  }

  render() {
    const combinedFeatures = this.combinedFeatures();
    const  {parentProduct, relatedProduct, show} = this.props;

    if (show) {
      return (
        <div className = 'related-product-modal'>
          <div className = 'comparing' >Comparing</div>
          <div className = 'related-product-modal-content'>
            <div className = 'related-product-modal-table-header'>
              <table>
                <tbody>
                  <tr >
                    <th className="related-product-modal-col related-product-modal-col-1">{parentProduct.name}</th>
                    <th className="related-product-modal-col related-product-modal-col-2"></th>
                    <th className="related-product-modal-col related-product-modal-col-3">{relatedProduct.name}</th>
                  </tr>
                </tbody>
              </table>
            </div>
            <div className = 'related-product-modal-table-body'>
              <table>
                <tfoot className = 'related-product-modal-table-footer'>
                  <tr >
                    <button className = 'related-product-modal-close-button' onClick = {(e) => {this.onClose()}}>Close</button>
                  </tr>
                </tfoot>
                <tbody>
                  {combinedFeatures.map((feature) => {
                    return (<tr key = {feature.feature} >
                      <th className="related-product-modal-col related-product-modal-col-1">{feature.parentProductValue}</th>
                      <th className="related-product-modal-col mrelated-product-odal-col-2">{feature.feature}</th>
                      <th className="related-product-modal-col related-product-modal-col-3">{feature.relatedProductValue}</th>
                    </tr>)
                  })}
                </tbody>
              </table>
            </div>
          </div>
        </div>
      )
    } else {
      return null;
    }
  }
}

export default RelatedProductModal
