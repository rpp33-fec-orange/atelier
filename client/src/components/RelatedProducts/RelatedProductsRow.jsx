// import React from 'react';
// import {GoChevronLeft, GoChevronRight} from 'react-icons/go';
// import images from './StockImages.jsx';
// import RelatedProductCard from './ProductCard.jsx';
// import $ from 'jquery';
// import setRatingSummary from './HelperFunctions.js'

// const photoUnavailable = 'https://icon-library.com/images/no-picture-available-icon/no-picture-available-icon-1.jpg'

// class RelatedProductsRow extends React.Component {

//   constructor(props) {
//     super(props);
//     this.state = {
//       currentPosition: 0,
//       currentPositionIndex: 0
//     }
//     this.scroll = this.scroll.bind(this);
//     this.relatedProductsRatingSummary = this.relatedProductsRatingSummary.bind(this)
//   }

//   scroll(direction) {
//     let far = $( '.related-product-container' ).width()/4*direction;
//     let pos = $('.related-product-container').scrollLeft() + far;
//     $('.related-product-container').animate( { scrollLeft: pos }, 1000)
//   }

//   relatedProductsRatingSummary (relatedProducts) {
//     for (var i = 0; i < relatedProducts.length; i++) {
//       var num_Rating = setRatingSummary(relatedProducts[i].meta_ratings);
//       relatedProducts[i]['num_Rating'] = num_Rating[0];
//     }
//     return relatedProducts;
//   }

//   render () {

//     const {relatedProductsIds, parentProduct, handleProductChange, relatedProductsInfo, handleStateChange} = this.props;

//     var relatedProductsWithRatings = this.relatedProductsRatingSummary(relatedProductsInfo);

//     if (this.props.initialized) {
//       var DOMarray = relatedProductsWithRatings.map((relatedProduct) => {
//         return (
//             <RelatedProductCard parentProduct = {parentProduct} relatedProduct = {relatedProduct} currentPosition = {this.state.currentPosition} handleProductChange = {handleProductChange} handleStateChange = {handleStateChange} initialized = {this.props.initialized}/>
//         );
//       });
//       return (
//         <div className = 'related-products-container'>
//           <a className ='prev' onClick = {this.scroll.bind(null, -1)}>&#10094;</a>
//           <div className = 'products-only-container'>
//             {DOMarray}
//           </div>
//           <a className ='next' onClick = {this.scroll.bind(null, 1)}>&#10095;</a>
//         </div>
//       )
//     } else {
//       return (
//         <div id="loading">
//           ⇆ Loading...
//         </div>
//        )
//     }
//   }
// }

// export default RelatedProductsRow;







import React from 'react';
import {GoChevronLeft, GoChevronRight} from 'react-icons/go';
import images from './StockImages.jsx';
import RelatedProductCard from './ProductCard.jsx';
import $ from 'jquery';
import setRatingSummary from './HelperFunctions.js'

const photoUnavailable = 'https://icon-library.com/images/no-picture-available-icon/no-picture-available-icon-1.jpg'

class RelatedProductsRow extends React.Component {

  constructor(props) {
    super(props);
    this.state = {
      prevDisable: true,
      nextDisable: this.refs && this.refs.offsetWidth >= this.refs.scrollWidth ? true : false
     }
    // this.scroll = this.scroll.bind(this);
    this.relatedProductsRatingSummary = this.relatedProductsRatingSummary.bind(this);
    this.checkButtons = this.checkButtons.bind(this);
  }

  componentDidMount() {
    this.checkButtons(this.refs.offsetWidth, this.refs.scrollWidth);
  }

  checkButtons = (offsetWidthValue, scrollWidthValue) => {
    this.setState({
     prevDisable: this.refs.scrollLeft <= 0 ? true : false,
     nextDisable:
     this.refs.scrollLeft + offsetWidthValue >= scrollWidthValue ? true : false
    });
   };

  // handleCarousel(currentOffsetWidth, currentScrollWidth) {
  //   this.setState({
  //     prevHidden: this.refs.scrollLeft <= 0 ? true : false,
  //     nextHidden:
  //     this.refs.scrollLeft + currentOffsetWidth >= currentScrollWidth ? true : false
  //    });
  // }

  // scroll(direction) {
  //   let far = $( '.related-product-container' ).width()/4*direction;
  //   let pos = $('.related-product-container').scrollLeft() + far;
  //   $('.related-product-container').animate( { scrollLeft: pos }, 1000)
  // }

  relatedProductsRatingSummary (relatedProducts) {
    for (var i = 0; i < relatedProducts.length; i++) {
      var num_Rating = setRatingSummary(relatedProducts[i].meta_ratings);
      relatedProducts[i]['num_Rating'] = num_Rating[0];
    }
    return relatedProducts;
  }

  render () {

    const {relatedProductsIds, parentProduct, handleProductChange, relatedProductsInfo, handleStateChange} = this.props;
    const offsetWidthValue = this.refs.offsetWidth,
          scrollWidthValue = this.refs.scrollWidth;

    var relatedProductsWithRatings = this.relatedProductsRatingSummary(relatedProductsInfo);

    if (this.props.initialized) {
      var DOMarray = relatedProductsWithRatings.map((relatedProduct) => {
        return (
            <RelatedProductCard parentProduct = {parentProduct} relatedProduct = {relatedProduct} currentPosition = {this.state.currentPosition} handleProductChange = {handleProductChange} handleStateChange = {handleStateChange} initialized = {this.props.initialized}/>
        );
      });
      return (
        <div className = 'related-products-container' ref = {(el) => {this.refs  = el; console.log('el', el);}}>
                <div className={`btn prev ${this.state.prevDisable ? "disable" : ""}`} disabled={this.state.prevDisable}
        onClick={() => {
          this.refs.scrollLeft -= offsetWidthValue / 2;
          this.checkButtons(offsetWidthValue, scrollWidthValue);
        }}>{'<'}</div>
          <div className = 'products-only-container' >
            {DOMarray}
          </div>
          <div className={`btn next ${this.state.nextDisable ? "disable" : ""}`} disabled={this.state.nextDisable}
            onClick={() => {
              this.refs.scrollLeft += offsetWidthValue / 2;
              this.checkButtons(offsetWidthValue, scrollWidthValue);
            }}>{'>'}</div>
        </div>
      )
    } else {
      return (
        <div id="loading">
          ⇆ Loading...
        </div>
       )
    }
  }
}

export default RelatedProductsRow;