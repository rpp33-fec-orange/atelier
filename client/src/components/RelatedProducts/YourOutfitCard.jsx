import React from 'react';
import { RiInformationLine } from 'react-icons/ri';
import StarRating from './Ratings.jsx';
import { AiOutlineCloseCircle } from 'react-icons/ai'

const photoUnavailable = 'https://icon-library.com/images/no-picture-available-icon/no-picture-available-icon-1.jpg'

class YourOutfitCard extends React.Component {

  constructor(props) {
    super(props);
    this.state = {
    }
  }

  render() {
    const  { product, handleDelete} = this.props;
    return (
      <div className = "your-outfit-card">
        <AiOutlineCloseCircle className = 'toggle-button' onClick = {() => {this.props.handleDelete()}}/>
        <img className = 'your-outfit-image' src = {product.photos[0].url || photoUnavailable} alt = {product.name}/>
        <div className = 'your-outfit-category'>{product.category}</div>
        <div className = 'your-outfit-name'>{product.name}</div>
        <div className = 'your-outfit-price'>${product.original_price}</div>
        <div className = 'your-outfit-rating'>
          <StarRating num = {product.num_Rating}/>
        </div>
      </div>
    )
  }

}

export default YourOutfitCard;