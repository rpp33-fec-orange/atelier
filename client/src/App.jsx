import React from 'react';
import $ from 'jquery';
import ProductOverview from './components/ProductOverview/index.jsx';
import RatingsReviews from './components/RatingsReviews/RatingsReviews.jsx';
import QuestionsAnswers from './components/QuestionsAnswers/index.jsx';
import RelatedProducts from './components/RelatedProducts/index.jsx';

class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      id: '',
      initialized: false,
      rating: 0,
      currentStyle: null,
      yourOutfitArray: []
    }
    this.productsHandler = this.productsHandler.bind(this);
    this.handleRating = this.handleRating.bind(this);
    this.getRating = this.getRating.bind(this);
    this.currentStyleHandler = this.currentStyleHandler.bind(this);
    this.yourOutfitHandleClick = this.yourOutfitHandleClick.bind(this);
    this.handleProductChange = this.handleProductChange.bind(this);
  }

  productsHandler() {
    $.ajax({
      context: this,
      type: 'GET',
      url: '/products',
      success: function (success) {
        // console.log('app ajax GET success');
        this.setState({
          id: success[0].id,
          initialized: true
        });
      },
      error: function (error) {
        console.log('app ajax GET error: ', error);
      },
      contentType: "application/json",
    })
  }

  handleRating(string) {
    console.log('star rating', string)
  }

  componentDidMount() {
    this.productsHandler();
  }

  componentWillMount() {
    this.getRating();
  }

  getRating(rating) {
    this.setState({
      rating: rating
    });
  }

  currentStyleHandler(selectedStyle) {
    this.setState({
      currentStyle: selectedStyle
    }, () => {
      console.log('currentStyle changed')
    })
  }

  yourOutfitHandleClick() {
    console.log('currentStyle', this.state.currentStyle)
    var currentStyle = this.state.currentStyle;
    currentStyle['num_Rating'] = this.state.rating;
    var yourOutfit = this.state.yourOutfitArray;
    const styleExists = yourOutfit.findIndex(element => element.style_id === currentStyle.style_id)
    if (styleExists === -1) {
      yourOutfit.push(currentStyle);
    }
    this.setState({
      yourOutfitArray: yourOutfit
    }, () => {
      console.log('outfit updated')
    })
  }

  handleProductChange(productId) {
    // console.log('id in app', productId)
    this.setState({ ...this.state, id: productId }, () => { console.log('product id updated') })
  }

  render() {

    if (this.state.initialized) {
      return (
        <div id="container">
          <ProductOverview id={this.state.id} rating={this.state.rating} currentStyleHandler={this.currentStyleHandler} yourOutfitHandleClick={this.yourOutfitHandleClick} />
          {/* <RelatedProducts id={this.state.id} yourOutfitArray={this.state.yourOutfitArray} yourOutfitHandleClick={this.yourOutfitHandleClick} handleProductChange={this.handleProductChange} />
          <QuestionsAnswers id={this.state.id} /> */}
          <RatingsReviews id={this.state.id} handleRating={this.handleRating} handleGetRating={this.getRating} />
        </div>
      )
    } else {
      return (
        <div id="loading">
          â‡† Loading...
        </div>
      )
    }
  }
}

export default App;
