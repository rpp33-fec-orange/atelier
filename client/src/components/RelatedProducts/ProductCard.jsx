import React from 'react';
import axios from 'axios';
import { RiInformationLine } from 'react-icons/ri';
import {AiOutlineStar} from 'react-icons/ai'
import RelatedProductModal from './RelatedProductModal.jsx';

const photoUnavailable = 'https://icon-library.com/images/no-picture-available-icon/no-picture-available-icon-1.jpg'

class RelatedProductCard extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      showModal: false
    }
    this.showModal = this.showModal.bind(this);
  }

  showModal(e) {
    this.setState({
      showModal: !this.state.showModal
    })
  }

  render() {
    const  { parentProduct, relatedProduct, currentPosition, handleProductChange } = this.props;
    // console.log('relatedProduct', relatedProduct)
    // console.log('parentProduct', parentProduct)
    return (
        <div className = "related-product-card" onClick = {() => {handleProductChange(relatedProduct.id)}}>
            <AiOutlineStar className = 'toggle-button' onClick = {(e) => {this.showModal()}}/>
            <RelatedProductModal parentProduct = {parentProduct} relatedProduct = {relatedProduct} onClose = {this.showModal} show = {this.state.showModal}/>
            <img className = 'related-product-image' src = {relatedProduct.photos[0].url || photoUnavailable} alt = {relatedProduct.name}/>
            <div className = 'related-product-category'>{relatedProduct.category}</div>
            <div className = 'related-product-name'>{relatedProduct.name}</div>
            <div className = 'related-product-price'>${relatedProduct.default_price}</div>
            <div className = 'related-product-rating'>★★★★☆</div>
        </div>
    )
  }

}

export default RelatedProductCard;