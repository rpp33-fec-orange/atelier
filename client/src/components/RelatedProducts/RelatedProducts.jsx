import React from 'react';
import RelatedProductsRow from './/relatedProdcutsRow.jsx'
import YourOutfitRow from './/yourOutfitRow.jsx'

const RelatedProducts = (props) => (
  <div>
    <h4 id = 'related-products'>Related Products</h4>
    <RelatedProductsRow/>
    <h4 id = 'related-products'>Your Outfit</h4>
    <YourOutfitRow/>
  </div>
)

export default RelatedProducts;