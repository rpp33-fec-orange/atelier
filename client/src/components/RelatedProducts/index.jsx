import React from 'react';
import RelatedProductsRow from './RelatedProductsRow.jsx'
import YourOutfitRow from './YourOutfitRow.jsx'
import $ from 'jquery';

class RelatedProducts extends React.Component {

  constructor(props) {
    super(props);
    this.state = {
      product_id: null,
      relatedProducts: [],
      relatedProductsInfo: [],
      relatedStylesInfo: [],
      parentProduct: [],
      relatedProductsIds: [],
      initialized: false
    };
    this.fetchParentProductData = this.fetchParentProductData.bind(this);
    this.fetchParentProductStyles = this.fetchParentProductStyles.bind(this);
    this.fetchRelatedProductsData = this.fetchRelatedProductsData.bind(this);
  }

  fetchParentProductData () {
    // This AJAX request fetches product data for parent product (shown in product overview section) info
    $.ajax({
      context: this,
      type: 'GET',
      url: `/products/${this.props.id}`,
      contentType: "application/json",
      success: function (data) {
        var tempArray = [];
        tempArray.push(data)
        this.setState({
          parentProduct: JSON.parse(JSON.stringify(tempArray))
        }, () => {
          this.fetchParentProductStyles();
        })
      },
      error: function (error) {
        console.log('error in GET request', error);
      },
    })
  }

  fetchParentProductStyles () {
    // This AJAX request fetches styles data (including photos) for parent product (shown in product overview section) info
    $.ajax({
      context: this,
      type: 'GET',
      url: `/products/${this.props.id}/styles`,
      contentType: "application/json",
      success: function (data) {
        const parentProduct = this.state.parentProduct[0];
        parentProduct['styles'] = data.results;
        this.setState({
          parentProduct: parentProduct
        }, () => {
          this.fetchRelatedProductsData();
        })
      },
      error: function (error) {
        console.log('error in GET request', error);
      },
    })
  }

  fetchRelatedProductsData () {
    // This AJAX request fetches related products info
    $.ajax({
      context: this,
      type: 'GET',
      url: `/products/${this.props.id}/related`,
      contentType: "application/json",
      success: function (data) {
        var relatedProductsIds = data.shift();
          this.setState({
            relatedProductsInfo: data,
            relatedProductsIds: relatedProductsIds,
            initialized: true
          })
      },
      error: function (error) {
        console.log('error in GET request', error);
      },
    })
  }

  componentDidMount () {
    this.setState({
      product_id: this.props.id,
    }, () => {
      this.fetchParentProductData();
    });
  }

  componentDidUpdate(prevProps) {
    if (this.props.id !== prevProps.id) {
        this.setState({
          product_id: this.props.id,
          initialized: false
        }, () => {
          this.fetchParentProductData();
        });
    }
  }

  render() {

    if (this.state.initialized) {
      return (
        <div className = 'related-products-and-items'>
          <div className = "related-products-row">
            <div className = 'heading'>Related Products</div>
            <RelatedProductsRow relatedProductsIds = {this.state.relatedProductsIds} parentProduct = {this.state.parentProduct} relatedProductsInfo = {this.state.relatedProductsInfo} handleProductChange = {this.props.handleProductChange} initialized = {this.state.initialized}/>
          </div>
          <div className = "your-outfit-row">
            <div className = 'heading'>Your Outfit</div>
            <YourOutfitRow currentStyle = {this.props.currentStyle} yourOutfitArray = {this.props.yourOutfitArray} yourOutfitHandleClick = {this.props.yourOutfitHandleClick}/>
          </div>
        </div>
      )
    } else {
      return (
        <div id="loading">
          â‡† Loading...
        </div>
       )
    }

  }
}

export default RelatedProducts;