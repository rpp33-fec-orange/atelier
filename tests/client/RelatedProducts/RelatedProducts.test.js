import React from 'react';
import TestRenderer from 'react-test-renderer';
import Enzyme from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import {render, screen} from '@testing-library/react'
import { shallow } from 'enzyme';
import { mount } from 'enzyme';
import { spy } from 'sinon';
import App from '../../../client/src/App.jsx';
import RelatedProductsRow from '../../../client/src/components/RelatedProducts/RelatedProductsRow.jsx';
import YourOutfitRow from '../../../client/src/components/RelatedProducts/YourOutfitRow.jsx';
import RelatedProducts from '../../../client/src/components/RelatedProducts/index.jsx';
import ProductCard from '../../../client/src/components/RelatedProducts/ProductCard.jsx';
import RelatedProductModal from '../../../client/src/components/RelatedProducts/RelatedProductModal.jsx';
import YourOutfitCard from '../../../client/src/components/RelatedProducts/YourOutfitCard.jsx';

Enzyme.configure({ adapter: new Adapter() });

describe ('Related Products', () => {

  it('should check that Parent App component exists', () => {
    const component = shallow (<App/>);
    expect(component.exists()).toBe(true);
  });

  it('RelatedProducts should render and match snapshot', () => {
    const tree = TestRenderer.create(<RelatedProducts />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should check that Related Products Row component exists', () => {
    const component = shallow (<RelatedProductsRow/>);
    expect(component.exists()).toBe(true);
  });

  it('should check that Your Outfit Row component exists', () => {
    const component = shallow (<YourOutfitRow/>);
    expect(component.exists()).toBe(true);
  });

  it('should check that Product card exists', () => {
    const component = shallow (<ProductCard/>);
    expect(component.exists()).toBe(true);
  });

  it('should check that Your Outfit Card exists', () => {
    const component = shallow (<YourOutfitCard/>);
    expect(component.exists()).toBe(true);
  });

  it('should check that Related Product Modal exists', () => {
    const component = shallow (<RelatedProductModal/>);
    expect(component.exists()).toBe(true);
  });

})

