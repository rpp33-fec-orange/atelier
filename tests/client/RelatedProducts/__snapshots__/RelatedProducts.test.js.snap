// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Related Products RelatedProducts should render and match snapshot 1`] = `
<div
  id="loading"
>
  â‡† Loading...
</div>
`;
